### PR TITLE
Add register scenarios

### LINKED ISSUE #41: Add 'generic' methods such as waiters and asserts
Reusable methods

### PR DESCRIPTION
**Pull request by NadejdaClima**
Added Register flow (feature file, steps, page objects and elements enum)

Other changes included:

1. Base Classes: deleted BaseAction and BasePage and refactored classes extending them
2. BaseClass.java now includes Driver instantiation and PageFactory.initElements()
2. Added asserts and wait method to GenericActions class
3. Moved CategoryEnum from page_objects package to page_objects.enums package
4. Replaced classic getters with lombok Getters in HomePage.java
5. Fixed error in Configuration.properties url value

### CODE REVIEW

File: src/main/java/com/endava/soa_5/actions/GenericActions.java [hunk 1]
```diff
@@ -1,12 +1,43 @@
...
+                .ignoring(NoSuchElementException.class);
+        wait.until(ExpectedConditions.visibilityOf(element));
+    }
+
+    public void assertPageIsDisplayed(String title){
+        assertThat("Wrong page is displayed", driver.getTitle(), is(title));
+    }
+    public void assertElementIsDisplayed(String selector){
+        int size = driver.findElements(By.xpath(selector)).size();
+        assertThat("No elements with xpath: '" + selector + "' were found", size, not(0));
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/actions/GenericActions.java]: correct message please

Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5/actions/GenericActions.java]: Could you please be more specific to how the message should be corrected?

File: src/main/java/com/endava/soa_5/actions/GenericActions.java [hunk 2]
```diff
@@ -1,12 +1,43 @@
...
+    Logger log = LoggerFactory.getLogger(this.getClass());
+
+    public void clickOnElement(WebElement element){
         element.click();
     }
-    //TODO: add generic waiter method here;
-    //TODO: add some generic asserts here;
-}
+
+    public void waitForElement(WebElement element, int timeout, int checkFrequency) {
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/actions/GenericActions.java]: timeout and checkFrequency can be added in property file and can be removed from method arguments

Comment (en-nclima) [reply src/main/java/com/endava/soa_5/actions/GenericActions.java:28]: will be added in later PR's

Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5/actions/GenericActions.java]: done

File: src/main/java/com/endava/soa_5/actions/GenericActions.java [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+import java.time.Duration;
+import java.util.NoSuchElementException;
 
-public class GenericActions extends BaseAction {
-    public static void clickOnElement(WebElement element){
+public class GenericActions extends BaseClass {
+    Logger log = LoggerFactory.getLogger(this.getClass());
+
+    public void clickOnElement(WebElement element){
         element.click();
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/actions/GenericActions.java:26]: as we have already waiters, can we use it here ?

**Review by EugenS21 — APPROVED** [review]

File: src/main/java/com/endava/soa_5/actions/GenericActions.java [hunk 4]
```diff
@@ -1,10 +1,43 @@
...
     }
-}
+
+    public void waitForElement(WebElement element) {
+        Wait<WebDriver> wait = new FluentWait<>(driver)
+                .withTimeout(Duration.ofSeconds(Long.parseLong(propertyReader.getPropertyValue("waitTimeout"))))
```
Comment (MarcelMorozniuc) [inline src/main/java/com/endava/soa_5/actions/GenericActions.java]: change better to milliseconds both

File: src/main/resources/configuration/Configuration.properties [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
+url=http://demowebshop.tricentis.com/
 chromeDriverPath=
 geckoDriverPath=
 userName=soa_user
 password=Soa_user5!
 browser=chrome
-emptyParameter=
\ No newline at end of file
+emptyParameter=
+waitTimeout=15
```
Comment (MarcelMorozniuc) [inline src/main/resources/configuration/Configuration.properties]: don't forgot to change here either, if update in method to milliseconds

**Review by MarcelMorozniuc — APPROVED** [review]

File: src/main/java/com/endava/soa_5/actions/GenericActions.java (hunk 3 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5/actions/GenericActions.java:26]: done thank you

File: src/main/java/com/endava/soa_5/actions/GenericActions.java (hunk 4 above)
Comment (NadejdaClima) [reply src/main/java/com/endava/soa_5/actions/GenericActions.java]: done

File: src/main/resources/configuration/Configuration.properties (hunk 5 above)
Comment (NadejdaClima) [reply src/main/resources/configuration/Configuration.properties]: done

**Pull request merged**